packages/web-components/fast-foundation/src/select/select.ts

```diff
@@ -0,0 +1,224 @@
+
+    /**
+     * The open attribute
+     */
+    @attr({ attribute: "open", mode: "boolean" })
+    public open: boolean;
+    private openChanged(): void {
+        if (this.$fastController.isConnected) {
+            this.button.setAttribute("aria-expanded", `${this.open}`);
+
+            if (this.open) {
+                this.listbox.focusFirstSelectedOption();
+            }
+        }
+    }
+
+    /**
+     * The container for the indicator icon.
+     * @internal
+     */
+    @observable
+    public indicatorContainer: HTMLElement;
+
+    /**
+     * @internal
+     */
+    @observable
+    public defaultSlottedNodes: Node[];
+
+    /**
+     * @internal
+     */
+    @observable
+    public listbox: Listbox;
+
+    /**
+     * @internal
+     */
+    @observable
+    public options: Option[];
+
+    /**
+     * The value displayed on the button
+     * @public
+     */
+    @observable
+    public displayValue: string;
+
+    /**
+     * The button element that displays the selected value and triggers the visibility of the listbox.
+     * @public
+     */
+    @observable
+    public button: HTMLElement;
+
+    /**
+     * The slot that holds the contents of the button
+     * @public
+     */
+    @observable
+    public slottedButtonContainer: Node[];
+
+    /**
+     * Update the value and display value when the listbox emits a change event.
+     * @param e - Event object
+     */
+    private listboxChangeHandler = (e: Event): void => {
+        const captured = e.target as Option;
+        this.value = captured.value;
+    };
+
+    protected valueChanged(previous: string, next: string): void {
+        super.valueChanged(previous, next);
+        DOM.queueUpdate(() => {
+            const selectedOption = this.listbox.selectedOption;
+            this.displayValue = selectedOption.textContent || selectedOption.value;
+        });
+    }
+
+    private updateForm(): void {
+        const value = this.value ? this.value : null;
+        this.setFormValue(value, value);
+    }
+
+    public keydownHandler = (e: KeyboardEvent): void => {
+        super.keypressHandler(e);
+
+        const keyCode = e.key || e.key.charCodeAt(0);
+
+        if (keyCode === "Escape") {
+            this.open = false;
+            DOM.queueUpdate(() => this.button.focus());
+            return;
+        }
+
+        if ([" ", "Enter"].includes(`${keyCode}`)) {
+            e.preventDefault();
+            e.stopPropagation();
+
+            if (!this.disabled) {
+                this.open = !this.open;
+            }
+
+            if (this.open) {
+                this.listbox.focusFirstSelectedOption();
+            } else {
+                DOM.queueUpdate(() => this.button.focus());
+            }
+
+            return;
+        }
+
+        // Pass the event to the listbox
+        if (!e.defaultPrevented) {
+            DOM.queueUpdate(() => {
+                e.preventDefault();
+                this.listbox.dispatchEvent(e);
```

Comment:
I'm concerned re-dispatching a keyboard event to a child could lead to weirdness.  Does it get bubble to select parents before listbox handles it, what if a parent calls preventDefault on it?   Why not just set a "selected item" prop on the listbox or call a setter function directly?